internal/cmd/envs.go

```diff
@@ -62,8 +62,115 @@ func envsCommand() *cobra.Command {
 	}
 	lsCmd.Flags().StringVarP(&outputFmt, "output", "o", "human", "human | json")
 	cmd.AddCommand(lsCmd)
-	cmd.AddCommand(stopEnvCommand(&user))
 
+	var (
+		img      string
+		tag      string
+		cpuCores float32
+		memGB    float32
+		diskGB   int
+		gpus     int
+		services []string
+	)
+
+	editEnvCmd := &cobra.Command{
+		Use:    "edit",
+		Short:  "edit an existing environment owned by the active user.",
+		Args:   cobra.ExactArgs(1),
+		Hidden: true,
+		Long:   "Edit an existing environment owned by the active user.",
+		Example: `coder envs edit back-end-env
+# increase number of cores for back-end env
+coder envs edit back-end-env --cores 4
+
+# increase disk storage for back-end-env
+coder envs edit back-end-env --disk 20
+`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			client, err := newClient()
+			if err != nil {
+				return err
+			}
+			envs, err := getEnvs(cmd.Context(), client, user)
+			if err != nil {
+				return err
+			}
+			if len(envs) < 1 {
+				flog.Info("no environments found")
+				return nil
+			}
+
+			name := args[0]
+
+			for _, e := range envs {
+				if e.Name == name {
+					// We're explicitly ignoring these errors because if any of these
+					// fail we are left with the zero value for the corresponding numeric type.
+					cpuCores, _ = cmd.Flags().GetFloat32("cores")
+					memGB, _ = cmd.Flags().GetFloat32("mem")
+					diskGB, _ = cmd.Flags().GetInt("disk")
+					gpus, _ = cmd.Flags().GetInt("gpus")
+
+					var updateReq coder.UpdateEnvironmentReq
+
+					// If any of the flags have defaulted to zero-values, it implies the user does not wish to change that value.
+					// With that said, we can enforce this by reassigning the request field to the corresponding existing environment value.
+					if cpuCores == 0 {
+						updateReq.CPUCores = &e.CPUCores
+					} else {
+						updateReq.CPUCores = &cpuCores
+					}
+
+					if memGB == 0 {
+						updateReq.MemoryGB = &e.MemoryGB
+					} else {
+						updateReq.MemoryGB = &memGB
+					}
+
+					if diskGB == 0 {
+						updateReq.DiskGB = &e.DiskGB
+					} else {
+						updateReq.DiskGB = &diskGB
+					}
+
+					if gpus == 0 {
+						updateReq.GPUs = &e.GPUs
+					} else {
+						updateReq.GPUs = &gpus
+					}
+
+					if img == "" {
+						updateReq.ImageID = &e.ImageID
+					} else {
+						updateReq.ImageID = &img
+					}
+
+					if tag == "" {
+						updateReq.ImageTag = &e.ImageTag
+					} else {
+						updateReq.ImageTag = &tag
+					}
+
+					if err := client.EditEnvironment(cmd.Context(), e.ID, updateReq); err != nil {
+						flog.Error("failed to apply changes to environment: '%s'", name)
```

Comment:
No need to ever log an error like this _unless_ we're iterating and performing fail-able operations that we want handled at the call site. Otherwise, the error returned by this func will be propagated and logged.